spec/avrolution/tasks_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+describe "rake tasks" do
+  include_context "rake setup"
+
+  let(:task_path) { 'avrolution/rake/rails_avrolution' }
+
+  describe "register_schemas" do
+    let(:task_name) { 'avro:register_schemas' }
+    let(:schema_files) { ['app.avsc'] }
+    let(:register_schemas) { Avrolution::RegisterSchemas.new(schema_files) }
+
+    before do
+      @original_schemas = ENV['schemas']
+      ENV['schemas'] = schema_files.join(',')
+      allow(Avrolution::RegisterSchemas).to receive(:new).and_return(register_schemas)
+      allow(register_schemas).to receive(:call).and_call_original
+    end
+
+    after do
+      ENV['schemas'] = @original_schemas
+    end
+
+    it "dispatches to a RegisterSchemas instance" do
+      task.invoke
+
+      expect(register_schemas).to have_received(:call)
+    end
+  end
+
+  describe "check_compatibility" do
+    let(:task_name) { 'avro:check_compatibility' }
+    let(:compatibility_check) { Avrolution::CompatibilityCheck.new }
+
+    before do
+      allow(Avrolution::CompatibilityCheck).to receive(:new).and_return(compatibility_check)
+      allow(compatibility_check).to receive(:call).and_call_original
+    end
+
+    it "dispatches to a CompatibilityCheck instance" do
+      task.invoke
+
+      expect(compatibility_check).to have_received(:call)
+    end
+  end
+
+  describe "add_compatibility_break" do
+    let(:task_name) { 'avro:add_compatibility_break' }
+
+    before do
+      allow(Avrolution::CompatibilityBreaksFile).to receive(:add)
+      ENV['name'] = 'test-name'
```

Comment:
These environment variables should should be cleared in an `after` block. Alternatively you could use [climate_control](https://github.com/thoughtbot/climate_control) for easier management of environment variables in tests.